### PR TITLE
Account reset UI

### PR DESCRIPTION
**Pull request by MarcoPolo**
Adds UI for chatting with users with account resets during your chat conversation.

screen shot 2017-02-08 at 2 33 21 pm

You can go back and forth between resets, handles TLFFinalize notification.

@keybase/react-hackers

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
The commits dc664617ea32...ffc6712515fe introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Old_Profile_Reset_Notice-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1486593359705-dc664617ea32-ffc6712515fe-linux/Old_Profile_Reset_Notice-Normal-dc664617ea32.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1486593359705-dc664617ea32-ffc6712515fe-linux/Old_Profile_Reset_Notice-Normal-ffc6712515fe.png)  
   Old_Profile_Reset_Notice-Normal comparison
</summary>

File: shared/actions/chat.js [hunk 1]
```diff
@@ -1759,6 +1834,39 @@ function _threadIsCleared (originalAction: Action, checkAction: Action): boolean
...
+    yield call(_updateInbox, conversations[0])
+  }
+}
+
+function * _openConversation ({payload: {conversationIDKey}}: Constants.OpenConversation): SagaGenerator<any, any> {
+  const inbox = yield select(inboxSelector)
+  const validInbox = inbox.find(c => c.get('conversationIDKey') === conversationIDKey && c.get('validated'))
+  if (!validInbox) {
+    yield put(({type: 'chat:getInboxAndUnbox', payload: {conversationIDKey}}: Constants.GetInboxAndUnbox))
+    yield take(a => a.type === 'chat:updateInbox' && a.payload.conversation && a.payload.conversation.conversationIDKey === conversationIDKey)
```
Comment (chromakode) [inline shared/actions/chat.js]: The coupling here seems a too loose. What happens if there's an error and a `chat:updateInbox` event isn't immediately dispatched? Is there an alternative?

File: shared/chat/conversations-list/container.js [hunk 2]
```diff
@@ -20,9 +21,10 @@ class ConversationListContainer extends Component {
 export default connect(
   (state: TypedState, {routeSelected}) => ({
     conversationUnreadCounts: state.chat.get('conversationUnreadCounts'),
-    inbox: state.chat.get('inbox').filter(i => !i.isEmpty || i.youCreated),
+    inbox: state.chat.get('inbox').filter(i => !i.isEmpty || i.youCreated)
```
Comment (chromakode) [inline shared/chat/conversations-list/container.js]: :sheep: This filtering logic is getting pretty intense... probably time to move it out of this connect and into a helper

File: shared/actions/chat.js [hunk 3]
```diff
@@ -988,7 +1062,7 @@ function * _loadMoreMessages (action: LoadMoreMessages): SagaGenerator<any, any>
   const inboxConvo = yield select(_selectedInboxSelector, conversationIDKey)
   if (inboxConvo && !inboxConvo.validated) {
     __DEV__ && console.log('Bailing on not yet validated conversation')
-    return
+    // return
```
Comment (MarcoPolo) [inline shared/actions/chat.js]: Whoops! that was a test of some other stuff, thanks!

File: shared/actions/chat.js [hunk 4]
```diff
@@ -319,6 +320,53 @@ function _inboxConversationToInboxState (convo: ?ConversationLocal): ?InboxState
...
+}
+
+function _inboxConversationLocalToSupersedesState (convo: ?ChatTypes.ConversationLocal): Constants.SupersedesState {
+  // TODO deep supersedes checking
+  if (!convo || !convo.info || !convo.info.id || !convo.supersedes) {
+    return Map()
+  }
+
+  const conversationIDKey = conversationIDToKey(convo.info.id)
+  const supersedes = _toSupersedeInfo(conversationIDKey, (convo.supersedes || []))
```
Comment (MarcoPolo) [inline shared/actions/chat.js]: it's a flow issue...

File: shared/actions/chat.js [hunk 5]
```diff
@@ -1095,6 +1169,7 @@ function _maybeAddTimestamp (message: Message, prevMessage: Message): MaybeTimes
 
   // messageID 1 is an unhandled placeholder. We want to add a timestamp before
   // the first message, as well as between any two messages with long duration.
+  // $FlowIssue with checking messageID
```
Comment (MarcoPolo) [inline shared/actions/chat.js]: even if I'm checking if it's a supersedes this still fails :/

**Comment by keybase-ci-visdiff** [comment]
The commits dc664617ea32...43a1c3ec26c2 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Old_Profile_Reset_Notice-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487117512957-dc664617ea32-43a1c3ec26c2-linux/Old_Profile_Reset_Notice-Normal-dc664617ea32.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487117512957-dc664617ea32-43a1c3ec26c2-linux/Old_Profile_Reset_Notice-Normal-43a1c3ec26c2.png)  
   Old_Profile_Reset_Notice-Normal comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits e9038c9d12bb...10bcbc89fdf9 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Old_Profile_Reset_Notice-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487120139322-e9038c9d12bb-10bcbc89fdf9-linux/Old_Profile_Reset_Notice-Normal-e9038c9d12bb.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1487120139322-e9038c9d12bb-10bcbc89fdf9-linux/Old_Profile_Reset_Notice-Normal-10bcbc89fdf9.png)  
   Old_Profile_Reset_Notice-Normal comparison
</summary>

**Pull request merged**